x/fee_grant/types/periodic_fee.go

```diff
@@ -47,7 +47,7 @@ func (a *PeriodicFeeAllowance) Accept(fee sdk.Coins, blockTime time.Time, blockH
 // last PeriodReset (eg. if you always do one tx per day, it will always reset the same time)
 // If we are more then one period out (eg. no activity in a week), reset is one Period from the execution of this method
 func (a *PeriodicFeeAllowance) TryResetPeriod(blockTime time.Time, blockHeight int64) {
-	if !a.PeriodReset.IsZero() && !a.PeriodReset.IsExpired(blockTime, blockHeight) {
+	if !a.PeriodReset.IsZero() && !a.PeriodReset.IsExpired(&blockTime, blockHeight) {
```

Comment:
Don't use `IsZero()`. Instead, we should use:
```
if a.PeriodReset.Unix() > 0 && ...
```

Reply:
context: #8282

Reply:
`a.PeriodReset` is not the `Timestamp` here it is struct of `ExpiresAt` (Which has a method `IsZero` to check both time and height)

Reply:
@atheeshp may be we should handle both cases separately as there's some known issue with `time.IsZero()`.

Reply:
@atheeshp  - yes. I suggested to rename that method in `PeriodRequest` because it's confusing.